FabLabKasse/faucardPayment/dinterface/magpos.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+"""
+pymagpos -- MagnaCarta POS protocol (minimal robust implementation)
+"""
+
+import serial
+import codes
+import logging
+import time
+
+class MagposError(Exception):
+    """ Base Exception Class for this Module """
+    pass
+
+
+class ResponseError(MagposError):
+    """
+    ResponseError occur when the response to a command does not match the command's OK signal.
+    """
+    def __init__(self, function, code):
+        self.code = code
+        self.function = function
+        self.response = [code]
+
+    def store_rawdata(self, response):
+        """ Stores the raw response for evaluation purpose """
+        self.raw = response
+
+    def read_rawdata(self):
+        """
+        Returns the raw response data
+        :return: Raw response data
+        :rtype: list of ints
+        """
+        return self.raw
+
+
+    def __str__(self):
+        return ("[{0}] Unintended response received:{1}".format(self.function,
+                codes.desc.setdefault(self.code, self.code)))
+
+
+class TransactionError(MagposError):
+    """
+    TransactionError occur when the amount that has been decreased does not match the given amount
+    """
+    def __init__(self, card, old, new, amount):
+        self.card = card
+        self.old = float(old)/100
+        self.new = float(new)/100
+        self.amount = float(amount)/100
+
+    def __str__(self):
+        return "Difference in balance does not match the amount that should have been decreased.\

```

Comment:
I have checked the the file and the '\' character is followed by standard cl rf return. Seems like github does mistake it for an escape character (\ + cl)